### PR TITLE
[ENH] Change to OpenNeuro-py downloader for example

### PR DESCRIPTION
**Pull request by alexrockhill**
Addresses https://github.com/mne-tools/mne-python/issues/9078.

Instead of downloading a whole tar file that is stashed somewhere, the idea is to change to using `openneuro-py` to download example data files.

### CODE REVIEW

File: examples/plot_visualize_bad_epochs.py [hunk 1]
```diff
@@ -14,32 +14,23 @@
 # sphinx_gallery_thumbnail_number = 2
 
 # %%
-# First, we download the data from OpenfMRI. We will download the tarfile,
-# extract the necessary files and delete the tar from the disk
+# First, we download the data from OpenfMRI which is hosted on OpenNeuro.
+# We will do this using ``openneuro-py``.
```
Comment (jasmainak) [inline examples/plot_visualize_bad_epochs.py:18]: Could you add a line here how to install it using `pip` ?

**Review by jasmainak — APPROVED** [review]
Otherwise, looks good! Thank you for the contribution!

File: examples/plot_visualize_bad_epochs.py [hunk 2]
```diff
@@ -14,32 +14,23 @@
...
 
 import os
-import tarfile
-
+import openneuro as on
```
Comment (sappelhoff) [inline examples/plot_visualize_bad_epochs.py:21]: I would not start with these abbreviations that nobody knows :)

just import openneuro and accept that it'll be a slightly longer line ... but then you won't have to look back to the imports every time you wonder what `on.xxx` means

File: examples/plot_visualize_bad_epochs.py [hunk 3]
```diff
@@ -14,32 +14,23 @@
...
-    print('Extracting files. This may take a while ...')
-    tf.extractall(path=base_path, members=tf.getmembers()[-25:-9:3])
-    os.remove(target)
+target_dir = os.path.join(
+    os.path.dirname(autoreject.__file__), '..', 'examples', dataset)
+if not os.path.isdir(target_dir):
+    os.makedirs(target_dir)
+
+on.download(dataset=dataset, target_dir=target_dir,
+            include=[f'sub-{subject_id}/ses-meg/'])
```
Comment (sappelhoff) [inline examples/plot_visualize_bad_epochs.py:33]:
```suggestion
openneuro.download(dataset=dataset, target_dir=target_dir,
                   include=[f'sub-{subject_id}/ses-meg/'])
```

File: examples/plot_visualize_bad_epochs.py (hunk 1 above)
Comment (sappelhoff) [reply examples/plot_visualize_bad_epochs.py:18]: also, it should be added as an optional dependency to https://github.com/autoreject/autoreject/blob/master/README.rst

**Review by sappelhoff — COMMENTED** [review]
this is really clean and nice :)

apart from the comments above, could you please check WHERE the data directory gets downloaded and adjust the caching in circleci?

See: https://github.com/autoreject/autoreject/blob/20e5b346bf85eb63a1db6b5134b4f8227d3694ed/.circleci/config.yml#L30-L35

previously, the data were saved in a directory `$CIRCLE_WORKING_DIRECTORY/examples/ds117`, which is your local path to autoreject, and then `./examples/ds117`

**Comment by alexrockhill** [comment]
I think I did the cache right but I'm not 100% sure...

**Review by sappelhoff — APPROVED** [review]
LGTM!

**Comment by alexrockhill** [comment]
The documentation built in under an hour the first commit and now it's almost two hours so I think it cached

**Comment by sappelhoff** [comment]
> The documentation built in under an hour the first commit and now it's almost two hours so I think it cached

yes, the doc build is very slow --> but that's a parallel issue to the data downloading ... it's always been a problem AFAIK, not recently introduced :) dunno how to fix it.

**Comment by jasmainak** [comment]
Now the only issue is that this dataset is not the same as the original. Seems to have been cleaned up somehow: https://747-59485140-gh.circle-artifacts.com/0/html/auto_examples/plot_visualize_bad_epochs.html#sphx-glr-auto-examples-plot-visualize-bad-epochs-py

I need to look more closely ...

**Comment by alexrockhill** [comment]
> Now the only issue is that this dataset is not the same as the original. Seems to have been cleaned up somehow: https://747-59485140-gh.circle-artifacts.com/0/html/auto_examples/plot_visualize_bad_epochs.html#sphx-glr-auto-examples-plot-visualize-bad-epochs-py
> 
> I need to look more closely ...

Still looks like the algorithm worked pretty well though...

**Comment by jasmainak** [comment]
@dengemann any opinion?

**Comment by jasmainak** [comment]
I remember spending hours debugging stuff in `autoreject` with the older dataset with @dengemann ... it helped us assess reasonability to some extent when tests were not covering everything.

**Comment by alexrockhill** [comment]
Hmmm, this seems reasonable, anything we can do to look more in-depth at if it is or not? I think it serves the purpose of this example so maybe that could be quantified in a test or another example and that might be helpful.

**Comment by jasmainak** [comment]
Let's wait for more opinions :)

**Comment by agramfort** [comment]
thanks heaps @alexrockhill !

**Pull request merged**